crates/gen/src/windows.rs

```diff
@@ -0,0 +1,26 @@
+/// Returns the build's `.windows` directory in the root of the workspace as a `PathBuf`.
+pub fn build_windows_dir() -> std::path::PathBuf {
+    let mut path = workspace_dir();
+    path.push(".windows");
+    path
+}
+
+fn workspace_dir() -> std::path::PathBuf {
+    // This is all rather complicated and expensive because Cargo lacks an environment variable for the workspace dir.
```

Comment:
Might want to consider using a `lazy_static!` or just `Once` like you have been to cache this.